### PR TITLE
Feat/qc 5371 format emails

### PR DESCRIPTION
**Pull request by thomaskosciuch**

### CODE REVIEW

File: cdk_build/DigitFtpStack.py [hunk 1]
```diff
@@ -68,32 +70,3 @@ def __init__(self, scope: Construct, construct_id='digitNbinTransfer', **kwargs)
         rule = self.cron(aws_events, self.cron_schedule,)
         rule.add_target(aws_events_targets.LambdaFunction(
             digit_nbin_transfer, retry_attempts=10))
-
```
Comment (QW-Jordan) [inline cdk_build/DigitFtpStack.py:71]: 🙏

File: digitNbinTransfer/compare_files.py
Comment (QW-Jordan) [inline digitNbinTransfer/compare_files.py:1]: what a better name!

File: digitNbinTransfer/failed_email.html [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+    <meta name="x-apple-disable-message-reformatting">  <!-- Disable auto-scale in iOS 10 Mail entirely -->
+    <title></title> <!-- The title tag shows in email notifications, like Android 4.4. -->
+
+    <link href="https://fonts.googleapis.com/css?family=Work+Sans:200,300,400,500,600,700" rel="stylesheet">
+
+    <!-- CSS Reset : BEGIN -->
+    <style>
+
+        html,
+body {
```
Comment (QW-Jordan) [inline digitNbinTransfer/failed_email.html:16]: this is giving me JAM flashbacks

File: digitNbinTransfer/main.py [hunk 3]
```diff
@@ -72,10 +76,31 @@ def older_than(source_time:int , destination_time:int) -> bool:
             files_to_add.append(source_file)
         else:
             files_to_add.append(source_file)
+    
+    if len(files_to_add) == 0:
+        return
+
 
+    def get_length() -> int:
```
Comment (QW-Jordan) [inline digitNbinTransfer/main.py]: why do you need this? what defines that length is always 1?

File: digitNbinTransfer/send_email.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    Body: SesContentBlock
+    Html: SesContentBlock
+
+class SesMessage(TypedDict):
+    Subject: SesContentBlock
+    Body: SesContentBlock
+
+local_datetime = datetime.now(timezone('America/Toronto'))
+
+def format_successful_email(n_transactions, destination) -> SesMessage:
```
Comment (QW-Jordan) [inline digitNbinTransfer/send_email.py]: what types are these?

File: requirements.txt
Comment (QW-Jordan) [inline requirements.txt:1]: why do we have 2 requirements files?

Comment (thomaskosciuch) [reply requirements.txt:1]: Great question!!

One of the requirements file is what's required to run the project, the other is what's required to run the lambda.

the lambda inherits the inner- requirement file

the other requires things that make the infra-as-code work PLUS all of the other things required to develop on this project

File: digitNbinTransfer/local_typing.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+class DigitCredentials(TypedDict):
```
Comment (thomaskosciuch) [inline digitNbinTransfer/local_typing.py:1]: moved all these here

File: digitNbinTransfer/sftp_connections.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+
+from os import environ
+from io import StringIO
+
+from paramiko import RSAKey
+from pysftp import Connection, CnOpts
+
+def nbin_sftp_connection() -> Connection:
```
Comment (thomaskosciuch) [inline digitNbinTransfer/sftp_connections.py:8]: moved these to their own docs

**Review by QW-Jordan — APPROVED** [review]

File: digitNbinTransfer/failed_email.html (hunk 2 above)
Comment (thomaskosciuch) [reply digitNbinTransfer/failed_email.html:16]: lol damn, i never got to do that stuff! I just found a rando template online and edited it x.x

**Pull request merged**